app/furniture/marketplace/marketplace_component.rb

```diff
@@ -0,0 +1,27 @@
+class Marketplace
+  class MarketplaceComponent < ApplicationComponent
+    attr_accessor :marketplace
+    delegate :location, :carts, :ready_for_shopping?, to: :marketplace
+
+    def initialize(marketplace:, data: {}, classes: "", **kwargs)
+      super(data: data, classes: classes, **kwargs)
```

Comment:
I think you could just leave the `data` and `classes` parameters to be part of `**kwargs` and not have to deal with them explicitly.

Relatedly, recently I came across `dry-initializer`, which the Evil Martians folks seem to like to use for their ViewComponents, and I am wondering if that would be a good way for us to tighten up our component constructors as well:
https://dry-rb.org/gems/dry-initializer/3.0/inheritance/
https://github.com/palkan/view_component-contrib#hanging-initialize-out-to-dry

Reply:
Ooo, I do like a lot of the the Evil Martians gems! I'd be down to give it a shot!